### PR TITLE
App: NewLocAppError -> NewAppError

### PR DESCRIPTION
**Pull request by grundleborg**
NewLocAppError -> NewAppError for the `app` package.

Sorry about the length of the diff :'(

### CODE REVIEW

File: app/ldap.go [hunk 1]
```diff
@@ -31,7 +31,7 @@ func TestLdap() *model.AppError {
 			return err
 		}
 	} else {
-		err := model.NewLocAppError("TestLdap", "ent.ldap.disabled.app_error", nil, "")
+		err := model.NewAppError("TestLdap", "ent.ldap.disabled.app_error", nil, "", http.StatusNotImplemented)
 		err.StatusCode = http.StatusNotImplemented
```
Comment (jwilander) [inline app/ldap.go]: Missed deleting this line :)

Comment (grundleborg) [reply app/ldap.go]: fixed

File: app/ldap.go [hunk 2]
```diff
@@ -18,7 +18,7 @@ func SyncLdap() {
 			if ldapI := einterfaces.GetLdapInterface(); ldapI != nil {
 				ldapI.SyncNow()
 			} else {
-				l4g.Error("%v", model.NewLocAppError("SyncLdap", "ent.ldap.disabled.app_error", nil, "").Error())
+				l4g.Error("%v", model.NewAppError("SyncLdap", "ent.ldap.disabled.app_error", nil, "", http.StatusNotImplemented).Error())
```
Comment (hmhealey) [inline app/ldap.go]: Not that you need to change it now since it always worked like this, but why did it make an error and immediately print it :p

**Comment by hmhealey** [comment]
@grundleborg Do any changes need to be made in the API documentation if any error codes have changed?

**Pull request merged**